### PR TITLE
adds samtools/bcftools/htslib 1.16

### PR DESCRIPTION
**Pull request by kapsakcj**
This PR adds a dockerfile and README.md for samtools, bcftools, and htslib 1.16

It also updates the main README.md with the new version number.

All dockerfiles are nearly identical to the previous versions, little to no changes made. 

All docker images build and tested successfully locally, and the built-in test layers pass.

`htslib/1.16/README.md` is a new file, I just modeled it off the others. Happy to hear feedback if there is any

This PR will address issue #437 

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

File: htslib/1.16/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+## Example Usage
+
+```bash
+# determine file formats for various BAM and SAM files
+$ htsfile tests/SARS-CoV-2/SRR13957123.primertrim.sorted.bam 
+tests/SARS-CoV-2/SRR13957123.primertrim.sorted.bam:     BAM version 1 compressed sequence data
+
+$ htsfile ce_tag_padded.sam
+ce_tag_padded.sam:      SAM version 1.4 sequence text
```
Comment (erinyoung) [inline htslib/1.16/README.md:20]: I just now realized I forgot to include a readme for 1.15. Could you add

```
# compresses sample.fastq to sample.fastq.gz
$ bgzip sample.fastq
```

to the htslit/1.16/README.md? It's what I mainly use this image for.

**Review by erinyoung — APPROVED** [review]
I approve of these containers, but I have a small request that fixes one of my prior mistakes:

Could you add bgzip to the htslib readme?

File: htslib/1.16/README.md (hunk 1 above)
Comment (kapsakcj) [reply htslib/1.16/README.md:20]: Sure!

**Comment by kapsakcj** [comment]
done 👍

**Review by erinyoung — APPROVED** [review]

**Comment by erinyoung** [comment]
Thank you!

I'm going to merge this and push the newest dockerfiles to dockerhub and quay

**Pull request merged**